src/main/java/com/vesta/config/security/JWTFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.vesta.config.security;
+
+import com.vesta.service.TokenService;
+import com.vesta.service.UserService;
+import com.vesta.service.converter.UserConverter;
+import com.vesta.service.dto.AuthentificationCredential;
+import com.vesta.service.dto.UserDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+public class JWTFilter extends GenericFilter {
+
+    @Autowired
+    private TokenService tokenService;
+
+    @Autowired
+    private UserService userService;
+
+    public JWTFilter(TokenService tokenService) {
+
+        this.tokenService = tokenService;
```

Comment:
Why you do this `this.tokenService = tokenService` ?  when you already have 
```
@Autowired
private TokenService tokenService;
```
Use only one, @Autowired or this.tokenService = tokenService;